### PR TITLE
Update reason dependency

### PR DESCRIPTION
**Pull request by pedrobslisboa**
## Description

This PR updates the reason to 3.11.0 and higher and formats the files.
It attempts to solve the format issues on CI that required us to fix the reason to `3.10.0`.

@davesnx after I cleaned the opam cache and update to `3.11.0` I was able to run `make format` and see the output we have in this PR.

I have to say that one of the changes was strange to me, the change to raw:
```diff
- let%browser_only mockInitWebsocket = () => [%mel.raw
+ let%browser_only mockInitWebsocket = () =>
+  {%mel.raw |
    function mockInitWebsocket() {
      console.log("Load JS");
    }
- |}
- ];
+ |};
```
I don't know if we want to have or change this in some way.

## Why to have this PR?

In theory, this PR will let us continue without the reason dep blocked. 
And then let us, for example, to update `server-reason-react` on `styled-ppx`
Captura de Tela 2024-11-04 às 19 59 39

### CODE REVIEW

File: demo/universal/native/lib/MelRaw.re [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
     console.log("Load JS");
   }
-|}
-];
+|};
 
-let%browser_only initWebsocket = () => [%mel.raw
-  {|
+let%browser_only initWebsocket = () =>
+  {%mel.raw |
```
Comment (davesnx) [inline demo/universal/native/lib/MelRaw.re:9]: Let's wait to merge this PR, until this is release: https://github.com/reasonml/reason/pull/2809

Comment (pedrobslisboa) [reply demo/universal/native/lib/MelRaw.re:9]: Excellent, I found it weird as well.

Comment (pedrobslisboa) [reply demo/universal/native/lib/MelRaw.re:9]: @davesnx Can we pin it until the release?

Comment (davesnx) [reply demo/universal/native/lib/MelRaw.re:9]: Sure, there are no pins in this repo but go for it

Comment (pedrobslisboa) [reply demo/universal/native/lib/MelRaw.re:9]: For anyone reading this, we decided to wait for the update.

**Pull request merged**